packages/backend/server/src/data/migrations/1699005339766-page-permission.ts

```diff
@@ -0,0 +1,87 @@
+import { PrismaService } from '../../prisma';
+
+export class PagePermission1699005339766 {
+  // do the migration
+  static async up(db: PrismaService) {
+    const turn = 0;
+    const lastTurnCount = 50;
+    const done = new Set<string>();
+
+    while (lastTurnCount === 50) {
+      const workspaces = await db.workspace.findMany({
+        skip: turn * 50,
+        take: 50,
+        orderBy: {
+          createdAt: 'asc',
+        },
+      });
+
+      for (const workspace of workspaces) {
+        if (done.has(workspace.id)) {
+          continue;
+        }
+
+        const oldPermissions =
+          await db.deprecatedUserWorkspacePermission.findMany({
+            where: {
+              workspaceId: workspace.id,
+            },
+          });
+
+        for (const oldPermission of oldPermissions) {
+          // mark subpage public
+          if (oldPermission.subPageId) {
+            const existed = await db.workspacePage.findUnique({
+              where: {
+                workspaceId_pageId: {
+                  workspaceId: oldPermission.workspaceId,
+                  pageId: oldPermission.subPageId,
+                },
+              },
+            });
+            if (!existed) {
+              await db.workspacePage.create({
+                select: null,
+                data: {
+                  workspaceId: oldPermission.workspaceId,
+                  pageId: oldPermission.subPageId,
+                  public: true,
+                },
+              });
+            }
```

Comment:
missing migration for single page share to a specific user in `WorkspacePageUserPermission`?